src/main/java/com/bravo/user/service/AddressService.java

```diff
@@ -0,0 +1,33 @@
+package com.bravo.user.service;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.bravo.user.dao.model.Address;
+import com.bravo.user.dao.model.mapper.ResourceMapper;
+import com.bravo.user.dao.repository.AddressRepository;
+import com.bravo.user.model.dto.AddressDto;
+
+@Service
+public class AddressService {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(AddressService.class);
+
+	private final AddressRepository addressRepository;
+	private final ResourceMapper resourceMapper;
+
+	public AddressService(AddressRepository addressRepository, ResourceMapper resourceMapper) {
+		this.addressRepository = addressRepository;
+		this.resourceMapper = resourceMapper;
+	}
+
+	public List<AddressDto> retrieveByUserId(final String userId) {
+		List<Address> addressList = addressRepository.findByUserId(userId);
```

Comment:
`addressList` can be `final` here.

Reply:
Does final on a list make it immutable -- for Kyle?

Reply:
@wheelerswebservices - in my past experience, I haven't been so liberal with specifying variables final, but after some research it sounds like this is beneficial in a few ways. IDEs are even able to implement missing keywords automatically as a Save action- thanks for the advice

@tlynema-bravolt - no, making a List (or any object) final does not make it's contents immutable, it simply locks in the reference